pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -133,17 +133,17 @@ public T get(String key) {
 
     @Override
     public Set<Map.Entry<String, T>> entrySet() {
-       return data.entrySet();
+       return new HashSet<>(data.entrySet());
```

Comment:
Instead of making a copy, it could be better to wrap it with `Collections.unmodifiableSet()` to avoid duplicating the entire map.

Reply:
Ah, ok.

Reply:
Even better: we could keep an `Collections.unmodifiableMap()` as a member variable and use it for these methods: 

```java
public Set<Map.Entry<String, T>> entrySet() {
    return readOnlyData.entrySet();
}
```

Reply:
fixed.

Reply:
> Collections.unmodifiableMap()

Ok, got it. Thanks.

Reply:
> Even better: we could keep an `Collections.unmodifiableMap()` as a member variable and use it for these methods:
> 
> ```java
> public Set<Map.Entry<String, T>> entrySet() {
>     return readOnlyData.entrySet();
> }
> ```

done